client/signup/steps/domains/index.jsx

```diff
@@ -655,6 +655,23 @@ class DomainsStep extends React.Component {
 			backLabelText = translate( 'Back to My Sites' );
 		}
 
+		// Override Back link if source parameter is found below
+		const backUrlSourceOverrides = {
+			'business-name-generator': '/business-name-generator',
+			domains: '/domains',
+		};
+		const source = get( this.props, 'queryObject.source' );
+
+		let isExternalBackUrl;
+
+		if ( backUrlSourceOverrides[ source ] ) {
```

Comment:
I'd add a check if `source` is present at all. TIL that this works:
```
aaa = { null: 123, };
console.log( aaa[ null ] ); // prints out 123
```
That's JS for you 😅 But to be on the safe side, I think an explicit check makes it clearer.

Reply:
JS keeps surprising me too :D, but I'm also in favour of explicit checks.